tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -279,3 +280,166 @@ def test_45498(self):
+
+        LOGGER.info("Step 2: Start IAM user CRUD operations while performing soft failure on "
+                    "random control pods in loop. Verify existing IAM users persist.")
+        LOGGER.info("Scale replicas for control pod to %s", self.repl_num)
+        resp = self.node_master_list[0].create_pod_replicas(num_replica=self.repl_num,
+                                                            deploy=self.deply_name)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Get the list of control pods")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        LOGGER.info("Total number of control pods in the cluster are: %s", self.repl_num)
+
+        iam_output = Queue()
+        event = threading.Event()
+        args = {'user_crud': True, 'num_users': self.num_users,
+                'output': iam_output, 'del_users_dict': del_users, 'header': self.header}
+        thr_iam = threading.Thread(target=self.ha_obj.iam_bucket_cruds,
+                                   args=(event,), kwargs=args)
+        thr_iam.daemon = True  # Daemonize thread
+        LOGGER.info("Starting background threads for IAM user creation/deletion.")
+        thr_iam.start()
+        for failure in range(HA_CFG["common_params"]["short_loop"]):
+            pod = secrets.SystemRandom().choice(pod_list)
+            LOGGER.info("Creating soft failure of %s for % loop", pod, failure)
+            self.node_master_list[0].kill_process_in_container(pod_name=pod,
+                                                               container_name=const.CORTX_CSM_POD,
+                                                               process_name=const.CSM_AGENT_PRC,
+                                                               safe_kill=True)
+            LOGGER.info("Step 2.1: Check cluster has failures.")
+            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+            assert_utils.assert_false(resp[0], resp)
+            LOGGER.info("Step 2.1: Checked cluster has failures.")
+        LOGGER.info("Step 2: Started IAM user CRUD operations while performing soft failure on "
+                    "random control pods in loop. Verified existing IAM users persisted.")
+
+        time.sleep(self.csm_soft_delay)
+        LOGGER.info("Step 3: Verify background process for IAM user creation/deletion")
+        event.clear()
+        LOGGER.info("Waiting for background threads to join")
+        thr_iam.join()
+        iam_resp = tuple()
+        while len(iam_resp) != 4:
+            iam_resp = iam_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not iam_resp:
+            assert_utils.assert_true(False, "Background process failed to do IAM user "
+                                            "creation/deletion operations")
+        failed = iam_resp[1]
+        exp_failed = iam_resp[0]
+        user_del_failed = iam_resp[2]
+        created_users = iam_resp[3]
```

Comment:
We should add check on created_users as well, that at least few users should get created.